invenio_users_resources/records/api.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 TU Wien.
+#
+# Invenio-Users-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""API classes for user and group management in Invenio."""
+
+
+from invenio_accounts.proxies import current_datastore
+from invenio_db import db
+from invenio_records.dumpers import ElasticsearchDumper
+from invenio_records.systemfields import ConstantField, DictField, ModelField
+from invenio_records_resources.records.api import Record
+from invenio_records_resources.records.systemfields import IndexField
+
+from .models import UserAggregateModel, UserGroupAggregateModel
+
+
+def parse_user_data(user):
+    """Parse the user's information into a dictionary."""
+    data = {
+        "id": user.id,
+        "email": user.email,
+        "username": None,
+        "full_name": None,
+    }
+
+    if user.profile is not None:
+        data["full_name"] = user.profile.full_name
+        data["username"] = user.profile.username
+        # TODO populate more when we have extensible user profiles
+
+    return data
+
+
+def parse_role_data(role):
+    """Parse the role's information into a dictionary."""
+    data = {"id": role.id, "name": role.name, "description": role.description}
+    return data
+
+
+class UserAggregate(Record):
```

Comment:
You probably have to override the ``revision_id`` to compute the aggregated version counter.